src/posix/platform/infra_if.hpp

```diff
@@ -35,8 +35,10 @@
 #define OT_POSIX_PLATFORM_INFRA_IF_HPP_
 
 #include "openthread-posix-config.h"
+#include "openthread-posix-core-related-config.h"
```

Comment:
@Irving-cl, I understand the goal of making spinel_interface modules independent of the OT code (as a standalone library), but I'm not clear why we need to change infra_if, netif, udp, etc., so they don't include openthread-core-config.h. These modules are part of the platform layer and should be able to see this header. They shouldn't be part of the spinel library module used by a host-side implementation.